pkg/deployment/member/state.go

```diff
@@ -100,44 +111,81 @@ func (s *stateInspector) RefreshState(ctx context.Context, members api.Deploymen
 	defer s.lock.Unlock()
 
 	results := make([]State, len(members))
-
-	nctx, cancel := globals.GetGlobalTimeouts().ArangoDCheck().WithTimeout(ctx)
-	defer cancel()
-
 	members.ForEach(func(id int) {
+		ctxChild, cancel := globals.GetGlobalTimeouts().ArangoDCheck().WithTimeout(ctx)
+		defer cancel()
+
 		if members[id].Group.IsArangosync() {
-			results[id] = s.fetchArangosyncMemberState(nctx, members[id])
+			results[id] = s.fetchArangosyncMemberState(ctxChild, members[id])
 		} else {
-			results[id] = s.fetchServerMemberState(nctx, members[id])
+			results[id] = s.fetchServerMemberState(ctxChild, members[id])
 		}
 	})
 
-	gctx, cancel := globals.GetGlobalTimeouts().ArangoDCheck().WithTimeout(ctx)
-	defer cancel()
-
 	var cs State
 	var h Health
+	var cli driver.Client
 
-	c, err := s.client.GetDatabaseClient(ctx)
-	if err != nil {
-		cs.NotReachableErr = err
-	} else {
-		v, err := c.Version(gctx)
-		if err != nil {
-			cs.NotReachableErr = err
-		} else {
-			cs.Version = v
-		}
+	globals.GetGlobalTimeouts().ArangoDCheck().RunWithTimeout(ctx, func(ctxChild context.Context) error {
+		cli, cs.NotReachableErr = s.deployment.GetDatabaseClient(ctxChild)
+		return nil
+	})
 
-		hctx, cancel := globals.GetGlobalTimeouts().ArangoDCheck().WithTimeout(ctx)
-		defer cancel()
-		if cluster, err := c.Cluster(hctx); err != nil {
-			h.Error = err
-		} else {
-			if health, err := cluster.Health(hctx); err != nil {
-				h.Error = err
-			} else {
-				h.Members = health.Health
+	if cs.NotReachableErr == nil {
+		// Set version.
+		cs.NotReachableErr = globals.GetGlobalTimeouts().ArangoDCheck().RunWithTimeout(ctx, func(ctxChild context.Context) error {
+			v, err := cli.Version(ctxChild)
+			if err == nil {
+				cs.Version = v
+			}
+			return err
+		})
+
+		if s.deployment.GetMode() == api.DeploymentModeCluster {
+			// Fetch health only in cluster mode.
```

Comment:
In all modes - we should never connect via service

Reply:
fixed